core/src/nodes/collection.rs

```diff
@@ -280,6 +280,7 @@ impl<TInEvent, TOutEvent> CollectionStream<TInEvent, TOutEvent> {
         TOutEvent: Send + 'static,
         THandler::OutboundOpenInfo: Send + 'static,     // TODO: shouldn't be required?
         TMuxer: StreamMuxer + Send + Sync + 'static,  // TODO: Send + Sync + 'static shouldn't be required
+        TMuxer::Substream: fmt::Debug,
```

Comment:
Why do you need here and in all those other places below the `Debug` constraint?

Reply:
…or else the compiler yells at me:

```
error[E0277]: `<TMuxer as muxing::StreamMuxer>::Substream` doesn't implement `std::fmt::Debug`
   --> core/src/nodes/collection.rs:286:29
    |
286 |         let id = self.inner.add_reach_attempt(future, handler);
    |                             ^^^^^^^^^^^^^^^^^ `<TMuxer as muxing::StreamMuxer>::Substream` cannot be formatted using `{:?}` because it doesn't implement `std::fmt::Debug`
    |
    = help: the trait `std::fmt::Debug` is not implemented for `<TMuxer as muxing::StreamMuxer>::Substream`
    = help: consider adding a `where <TMuxer as muxing::StreamMuxer>::Substream: std::fmt::Debug` bound
```

Reply:
But only because [you added the constraint](https://github.com/libp2p/rust-libp2p/pull/562/files#diff-dea5f941e495f207489ed6e87c902662R143) `TMuxer::Substream: fmt::Debug` to `HandledNodesTasks::add_reach_attempt`. I was asking why you need this constraint here *and in all those other places below* (which includes the change to `HandledNodesTasks::add_reach_attempt`)? As far as I can see nothing makes use of `Debug`.

Reply:
Sorry, it seems that the propagation of `Debug` is caused by the existing derive of `Debug` on `NodeEvent` which holds a `SubstreamRef`.